third-party/crowdsec/src/main/java/org/apache/james/CrowdsecImapConnectionCheck.java

```diff
@@ -41,7 +41,7 @@ public CrowdsecImapConnectionCheck(CrowdsecClientConfiguration crowdsecClientCon
 
     @Override
     public Publisher<Void> validate(InetSocketAddress remoteAddress) {
-        String ip = remoteAddress.getAddress().getHostAddress();
+        String ip = remoteAddress.getHostName();
```

Comment:
Q: Are they the same?

Reply:
There is a bit difference.

```
InetSocketAddress.createUnresolved(haproxyMsg.sourceAddress(), haproxyMsg.sourcePort())
```

I created `InetSocketAddress` by this way, so addr will be null. If I call `getAddress`, it will throw NullPointerException. That's why I changed it
```
private InetAddress getAddress() {
    return addr;
}
```

Reply:
Wrong choice led to regression is my feeling about this...

Reply:
Yep, but one positive thing is that it looks more succinct :)

Reply:
I disagree with keeping this change. `succinct` has no point if it is a regression.

see:
- https://stackoverflow.com/questions/55521964/different-behavior-of-inetsocketaddressgethostname-in-windows-and-linux-for-12
- https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getHostName--

hostname is different from IP address. Be careful with what you are doing.